t_clickhouse/test_clickhouse_logs.py

```diff
@@ -0,0 +1,463 @@
+                    ("Referer", "https://somesite.com"),
+                ],
+            ),
+        )
+        self.assertTrue(self.dmesg_logs_exists())
+
+        stdout = self.get_access_log_file_data(self.get_tempesta())
+        self.assertTrue(stdout.endswith("Daemon started\n"))
+        self.assertEqual(self.clickhouse_logs.total_count(), 1)
+
+        record = self.clickhouse_logs.read()[0]
+        t1 = record.timestamp.replace(microsecond=0, second=0, tzinfo=timezone.utc)
+        t2 = datetime.now(tz=timezone.utc).replace(microsecond=0, second=0)
+        self.assertEqual(t1, t2)
+        self.assertEqual(
+            record.user_agent,
+            "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0",
+        )
+        self.assertEqual(record.uri, "/test")
+        self.assertEqual(record.address, IPv4Address(tf_cfg.cfg.get("Client", "ip")))
+        self.assertEqual(record.referer, "https://somesite.com")
+        self.assertEqual(record.status, 201)
+        self.assertEqual(record.version, 3)
+        self.assertEqual(record.method, 3)
+        self.assertEqual(record.response_content_length, 8)
+        self.assertEqual(record.dropped_events, 0)
+        self.assertEqual(record.vhost, "default")
+
+
+class TestClickHouseLogsCorrectnessDataPostRequest(TestClickhouseLogsBaseTest):
+    backends = [
+        {
+            "id": "deproxy",
+            "type": "deproxy",
+            "port": "8000",
+            "response": "static",
+            "response_content": "HTTP/1.1 500 Internal Server Error\r\n"
+            "Content-Length: 8\r\n\r\n12345678",
+        },
+    ]
+
+    def test_clickhouse_record_data_with_post(self):
+        """
+        Verify the clickhouse log record data
+        """
+        client = self.get_client("deproxy")
+        client.start()
+
+        time.sleep(1)
+        self.send_simple_request(
+            client, client.create_request(uri="/", method="POST", headers=[]), expected_status="500"
+        )
+
+        record = self.clickhouse_logs.read()[0]
+        self.assertEqual(record.status, 500)
+        self.assertEqual(record.method, 10)
+        self.assertEqual(record.response_content_length, 8)
+        self.assertEqual(record.dropped_events, 0)
+
+
+@unittest.skip("utc not default time zone")
+class TestClickHouseLogsDelay(TestClickhouseLogsBaseTest):
+    def test_correctness_time_of_logs_after_server_delay(self):
+        """
+        Verify the correctness of the clickhouse record
+        timestamp value and timezone, response time
+        """
+        server = self.get_server("deproxy")
+        server.sleep_when_receiving_data = 2
+
+        client = self.get_client("deproxy")
+        client.start()
+
+        time.sleep(1)
+        time_before = datetime.now(tz=timezone.utc)
+
+        self.send_simple_request(client)
+
+        time_after = datetime.now(tz=timezone.utc)
+        self.assertEqual((time_before - time_after).seconds, 2)
```

Comment:
I think this test can be unstable. You should use `assertGreater` here or remove this test.